testng-core/src/test/java/test/retryAnalyzer/github2669/IssueTest.java

```diff
@@ -0,0 +1,19 @@
+package test.retryAnalyzer.github2669;
+
+import org.testng.Assert;
+import org.testng.TestNG;
+import org.testng.annotations.Test;
+import test.SimpleBaseTest;
+import java.util.Collections;
+
+public class IssueTest extends SimpleBaseTest {
+  @Test
+  public void testFailedRetryWithParameters() {
+    TestNG testng = create();
+    String testngXmlPath = getPathToResource("xml/github2669.xml");
```

Comment:
Why not work with the XmlSuite and XmlTest instead? That would be a lot more easier with respect to keeping the details of the test within the confines of this test method itself and not have to worry about all these extra suite files.